File: Python/Greatest_number_from_string.py
import re


def greatest_number_from_mixed_string(text: str) -> int:
    numbers = [int(n) for n in re.findall(r"\d+", text)]
    if not numbers:
        raise ValueError("No numbers found in string.")
    return max(numbers)


def main() -> None:
    text = input("Enter mixed string: ").strip()

    try:
        result = greatest_number_from_mixed_string(text)
        print("Greatest number:", result)
    except ValueError as error:
        print("Error:", error)


if __name__ == "__main__":
    main()

File: Typescript/Greatest_number_from_string.ts
const readline = require('node:readline/promises');
const { stdin, stdout } = require('node:process');

const greatestNumberFromMixedString = (text) => {
  const numbers = (text.match(/\d+/g) || []).map(Number);

  if (numbers.length === 0) {
    throw new Error('No numbers found in string.');
  }

  return Math.max(...numbers);
};

const main = async () => {
  const rl = readline.createInterface({ input: stdin, output: stdout });

  try {
    const text = (await rl.question('Enter mixed string: ')).trim();
    const result = greatestNumberFromMixedString(text);
    console.log('Greatest number:', result);
  } catch (error) {
    console.log('Error:', error.message);
  } finally {
    rl.close();
  }
};

main();
